ci-operator/step-registry/ipi/conf/proxy/ipi-conf-proxy-commands.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+if [[ ! -f "${SHARED_DIR}/proxy_private_url" ]]; then
+  echo "'${SHARED_DIR}/proxy_private_url' not found, abort." && exit 1
+fi
+
+# TODO: move to image
+curl -L https://github.com/mikefarah/yq/releases/download/3.3.0/yq_linux_amd64 -o /tmp/yq && chmod +x /tmp/yq
+
+CONFIG="${SHARED_DIR}/install-config.yaml"
+PATCH=/tmp/install-config-proxy.yaml.patch
+
+proxy_private_url=$(< "${SHARED_DIR}/proxy_private_url")
+cat > "${PATCH}" << EOF
+proxy:
+  httpProxy: ${proxy_private_url}
+  httpsProxy: ${proxy_private_url}
```

Comment:
In `ci-operator/step-registry/ipi/conf/aws/proxy/ipi-conf-aws-proxy-commands.sh`, we set these up in the same step that creates the proxy, so there's no need to pass the proxy URI through `SHARED_DIR`.  Are there benefits to splitting the `proxy` install-config work into a separate step?

Reply:
Because proxy setting is a common setting on different platforms, I think introduce this simple step will benefit for other cloud providers,  we also introduced provision vpc/bastion/proxy in pre-actions, but not combine everything in `ipi-conf-{aws,azure,gcp}-proxy`, that will also help us easily place the proxy host in different cases.

Reply:
We even think about refactoring the existing `ipi-conf-aws-proxy` like this, but to avoid massive noise on the existing ci jobs, we did not touch with the existing `ipi-conf-aws-proxy`

Reply:
```sh
cat >> "${CONFIG}" << EOF
proxy:
  httpsProxy: ${PROXY_URL}
  httpProxy: ${PROXY_URL}
EOF
```

is not a lot of effort locally, and having something like that in each proxy-creating step seems like it might be easier than the overhead of a new step just to write those three lines into the install-config.  With [`HTTPS_PROXY` Git access fixed in 4.6][1], perhaps we can restore `additionalTrustBundle` for HTTPS proxy traffic, like we [used to have][2].  Or [maybe not][3]?

[1]: https://bugzilla.redhat.com/show_bug.cgi?id=1750650#c50
[2]: https://github.com/openshift/release/pull/5133/files#diff-a73b08887ec6ccd5778c1915a196dea5a79038c959f87b4fd1bb886ced4911c3R917
[3]: https://bugzilla.redhat.com/show_bug.cgi?id=1881790

Reply:
If only take `cluster behind http_proxy or https_proxy` into consideration, maybe `ipi-conf-aws-proxy` maybe more easier, but we also need to take other test scenarios into consideration.
Beside `cluster behind http_proxy or https_proxy`, we also need to test `private cluster`, in that case, all the cli clients also need a proxy to access the cluster, the proxy server setting up process is almost the same, if that case, we need to copy most of codes from `ipi-conf-aws-proxy` to create a new step. But the codes sounds a bit duplicated. The QE's resolution sounds more like extracting `proxy server setting up` as a separate step to avoid duplicates codes. 

`Creating several different steps for cases which requires proxy server` vs `Creating proxy server in one step + Consume it from SHARED_DIR for cases which requires proxy server`, personally I think the latter one is more easier and easier to maintain.

Even if we extend similar test cases (that need a proxy server) to more, e.g: `cluster install using mirror registry`, the mirror registry server setup process will hit similar issue, `Creating a dedicated step for cases which requires mirror registry` vs `Creating mirror registry server in one step + Consume it from SHARED_DIR for cases which requires mirror server`, if we choose the latter one, we even can combine the mirror registry + proxy in one step (the step of launching vm can be shared, prepare ignitions with both proxy and mirror registry configuration enabled).